README.md

```diff
@@ -151,6 +151,18 @@ where `getProperty` method is called from.
 Only non-native Java methods are supported. To profile a native method,
 use hardware breakpoint event instead, e.g. `-e Java_java_lang_Throwable_fillInStackTrace`
 
+**Be aware** that the first instrumentation of a non-native Java method may
+cause codecache flush invoked by ```VM_RedefineClasses::redefine_single_class``` ->
+```VM_RedefineClasses::flush_dependent_code```. After such an action, your application
+works at the interpreter, and the JIT compiler needs to compile every method from a scratch 
```

Comment:
I used method profiling a lot in our production environment and didn't notice any performance issues. Now, after looking into this deeper, I see that HotSpot indeed flushes all nmethods on the first class redifinition, but only when JVM TI agent is attached at runtime. When it is loaded at JVM startup (using `-agentpath/-agentlib` option), the dependencies are recorded from the very beginning, and thus there is no massive deoptimization on the first redifinition - that's exactly our case.

I agree, it makes sense to document the deoptimization problem, and mention the conditions under which it happens, since switching to the interpreter is not always the case.

Reply:
Thank you for a clarification, I changed it in my PR. I used to attach profiler from Java code with ```AsyncProfiler.getInstance()```, I need to consider switching to attach as an agent.